lib/modules/manager/puppet/constants.ts

```diff
@@ -0,0 +1,12 @@
+import { regEx } from '../../../util/regex';
+
+export const simpleModuleLineRegexFactory = (): RegExp =>
+  /^mod\s*'([^']+)',\s*'([^']+)'$/gm;
+export const gitModuleRegexFactory = (): RegExp =>
+  /^mod '(?<moduleName>[^']+)',(?![^\n])(?:\n\s+:(?<key1>\w+)\s+=>\s+'(?<value1>[^']+)',?)(?![^\n])(?:\n\s+:(?<key2>\w+)\s+=>\s+'(?<value2>[^']+)',?)?(?![^\n])(?:\n\s+:(?<key3>\w+)\s+=>\s+'(?<value3>[^']+)',?)?/gm;
```

Comment:
Only glancing at the regexp, not running it mysql, it seems to require newlines for certain keys?

```ruby
mod 'foo', :git => 'git@example.com', :branch => 'baz'
mod 'bar', git: 'git@example.com', branch: 'foo'
```
Unsure if those would match.

Reply:
you're right, it doesn't match: https://regexr.com/6i432

Reply:
I created a Puppetfile parser, that handles this case correctly.